lib/cli/options.js

```diff
@@ -110,6 +110,9 @@ const parse = (args = [], defaultValues = {}, ...configObjects) => {
   // 4. we can then reapply the values after yargs-parser is done.
   const nodeArgs = (Array.isArray(args) ? args : args.split(' ')).reduce(
     (acc, arg) => {
+      if (typeof arg !== 'string') {
+        throw new Error(`Invalid option ${arg} passed to mocha cli`);
```

Comment:
[Feature] If there are a lot of flags, it'd be nice to know _which_ is wrong. Also, in the case of a very long arg, it'd be easier to put the flag first:

```suggestion
        throw new Error(`Mocha flag '--${flag}' given invalid option: '${arg}'`);
```

(or something like that: I haven't run the code to check)